ruby_programming/git/lesson_a_deeper_look_at_git.md

```diff
@@ -79,41 +81,38 @@ Once you're satisfied with your changes, run
+Using `squash` for our commits is a very handy way of keeping our Git history tidy. It's important to know how to `squash`, because this process may be the standard on some development teams. Squashing makes it easier for others to understand the history of your project. What often happens when a feature is merged, is we end up with some visually complex logs of all the changes a feature branch had on a main branch. These commits are important while the feature is in development, but aren't really necessary when looking through the entire history of your main branch.
 
-Let's say I want to `squash` all three of the commits I previously worked with together into the first commit on the list, which is `Rewrite Introduction`. I would simply  `pick` that commit as the commit that the other two are being `squash`ed into:
+Let's say we want to `squash` the second commit into the first commit on the list, which is `Create first file`. First let's rebase all the way back to our root commit by typing `git rebase -i --root`. Now what we'll do is `pick` that first commit, as the one which the second commit is being `squash`ed into:
 
 ~~~bash
-pick c85c71dd5 Rewrite Introduction
-squash 4ad2c8df6 Add information on using amend
-squash 729db1da4 Begin work on Remotes
+pick e30ff48 Create first file
+squash 92aa6f3 Create second file
+pick 05e5413 Create third file and create fourth file
 ~~~
 
-That's it! When I save and exit the editor, the bottom two commits on that list get squashed into the top commit. 
+Rename the commit to `Create first and second file`, then finish the rebase with `git rebase --continue`. That's it! Run `git log` and see how the first two commits got squashed together. 
 
 #### Splitting Up a Commit
 
-Before diving into Remotes, we're going to have a look at a handy Git command called `reset`. Let's say one of my commits was `c85c71dd5 Add knight class and Add attack module`. That's a mouthful, and a bit much for one commit as you should have learned in the previous lesson on commits. So what we're going to do is split it up into two smaller commits by, once again, using the interactive `rebase` tool. 
+Before diving into Remotes, we're going to have a look at a handy Git command called `reset`. Let's consider the commits we have at the moment `Create third file and create fourth file`. That's a mouthful, and a bit much for one commit as you should have learned in the previous lesson on commits. So what we're going to do is split it up into two smaller commits by, once again, using the interactive `rebase` tool. 
```

Comment:
Since this is really just creating two files I wouldn't say it's a good example of having too much for a single commit, and I think that saying it is might lead some people to bad Git practices. I think it would be good for you to highlight that in this walkthrough we're just using individual blank files for convenience.

Reply:
Great point, thanks!